src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -488,6 +489,13 @@ extern "C" void PalAttachThread(void* thread)
 #endif
 }
 
+#ifdef TARGET_LINUX
+int __cxa_thread_atexit(void (*func)(void*), void* t, void* c)
+{
+    return __cxa_thread_atexit_impl(func, t, c);
```

Comment:
`__cxa_thread_atexit_impl` is available on glibc >= 2.18, but not older versions (used by centos7...) or musl-libc, so I will dynamically load the symb (with `dlsym` and co.).

But what should be the fallback implementation? call destructor under a pthread lock or a nop?

Reply:
My understanding is that the per-thread atexit is not implemented in those variants of libcs, it fallsbac to libstdc++ implementation or one provided by the compiler: https://reviews.llvm.org/D21803?id=62394.